### ISSUE TITLE
Avoid release builds from timing out

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#152186]
Long-term tracking issue for https://github.com/flutter/flutter/issues/151249.

TL;DR the issue is that there is less capacity in dart-internal (which engine release builds use) than flutter prod (which engine main branch builds use), which leads to longer queue times, which leads to builds which schedule and than collect sub-builds to timeout.

To prevent this in the future, we should:

- [x] Hard-code in the top-level release builders to have longer timeouts - the repo is internal, but these are already set to timeout after 5 hours: https://dart-internal.googlesource.com/dart-internal/+/refs/heads/main/flutter-internal/engine.star#57
- [x] edit the recipes--when scheduling sub-builds and on a release branch, to use a longer timeout https://flutter-review.googlesource.com/c/recipes/+/59360
- [ ] Increase bot capacity in dart-internal

**Comment by zanderso** [comment]
Capacity for dart-internal seems to be defined here: https://chrome-internal.googlesource.com/infradata/config/+/refs/heads/main/configs/chrome-swarming/starlark/bots/dart.star#279.

@athomas all of our release builds are timing out due to queueing esp. for those 5 kvm bots. Do we just need to increase the numbers in that config file to get more bots in the pool, or are there more steps?

**Comment by athomas** [comment]
You can just increase the number by some reasonable amount (say 5), if you want to add "many" we need to ensure we have quota for that.

**Comment by zanderso** [comment]
> You can just increase the number by some reasonable amount (say 5), if you want to add "many" we need to ensure we have quota for that.

Sent https://chrome-internal-review.googlesource.com/c/infradata/config/+/7590413

**Comment by zanderso** [comment]
https://chrome-internal-review.googlesource.com/c/infradata/config/+/7590413 landed, so we should have a bit more capacity, but we should still keep an eye open for long queue times. I expect we'll still see some.

**Comment by christopherfujino** [comment]
> https://chrome-internal-review.googlesource.com/c/infradata/config/+/7590413 landed, so we should have a bit more capacity, but we should still keep an eye open for long queue times. I expect we'll still see some.

Thanks! We should also be careful about building for multiple release branches concurrently. Also, we should be mindful of internal b/340303879.

**Comment by zanderso** [comment]
Part of the capacity issue is that the orchestrators aren't participating in the actual work, and are sitting on ~10 out of 30 of the 32 core Linux machines doing nothing on every build. We will get more capacity for free by fixing that problem.

**Comment by christopherfujino** [comment]
Looks like my CL to extend timeout of sub-builds on release branches worked: https://ci.chromium.org/ui/p/dart-internal/builders/flutter/Linux%20Production%20Engine%20Drone/39306/infra

**Comment by zanderso** [comment]
How is this going? Should we still be thinking about requesting more quota for more 32 core machines for the linux builders?

**Comment by christopherfujino** [comment]
The runtimes [here](https://ci.chromium.org/ui/p/dart-internal/builders/flutter/Linux%20engine_release_builder) since August 23 (when I landed my recipes CL to extend the timeout of sub-builds: https://flutter.googlesource.com/recipes/+/f3c6d3a659c964959260a02850633d2b8ea4df7e) has topped out at 1.4 hours, which seems reasonable (the only infra failure was an unrelated failure to cache from CAS, which passed on a re-run).

However, we haven't yet started branching for internal rolls, b/340303879, which I think is what will really stress test dart-internal.

**Comment by matanlurey** [comment]
Closing as not particularly actionable - we can open specific issues as the need arrises again.
